pkg/cloudprovider/events/events.go

```diff
@@ -40,3 +46,22 @@ func NodeClaimFailedToResolveNodeClass(nodeClaim *v1.NodeClaim) events.Event {
 		DedupeValues:   []string{string(nodeClaim.UID)},
 	}
 }
+
+func NodeClaimFailedToRegister(nodeClaim *v1.NodeClaim, err error) events.Event {
+	return events.Event{
+		InvolvedObject: nodeClaim,
+		Type:           corev1.EventTypeWarning,
+		Reason:         AsyncProvisioningReason, // TODO: our other events don't set reason... is it an oversight?
+		Message:        fmt.Sprintf("NodeClaim %s failed to register: %s", nodeClaim.Name, truncateMessage(err.Error())),
```

Comment:
Do we need to include nodeClaim.Name in message if it is recorded as InvolvedObject?

Reply:
It doesn't make sense to include nodeClaim name in this message, agreed!